metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py

```diff
@@ -846,41 +846,54 @@ def create_lineage(
                 return Lineage.empty()
 
         arguments = self.get_tokens(data_access_func_detail.arg_list)
-        if len(arguments) < 4:
-            logger.info(
-                f"Databricks workspace and catalog information in arguments({arguments}). "
-                f"Skipping upstream table"
-            )
-            return Lineage.empty()
 
-        workspace_fqdn: str = arguments[0]
+        if (
```

Comment:
Wouldn't it be better to move this logic into the get_tokens and return a typed class like 
```
@dataclass 
class ReferencedTable:
   warehouse: str 
   catalog: Optional[str]
   database: str
   table: str
```

If the get token returns an object like this, then I think you can remove logic from various places where you now have to remember the exact order of the token array.

Reply:
Thanks :)